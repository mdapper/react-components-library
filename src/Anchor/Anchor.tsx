import React, { ReactNode } from 'react';

type Props = {
  children: ReactNode;

  /**
   * Set this to change the background color
   * @default "blue"
   */
  backgroundColor?: string;

  /**
   * Set this to change the href
   * @default "#"
   */
  href?: string;
};

export const Anchor = ({
  children,
  backgroundColor = 'blue',
  href = '#',
  ...props
}: Props) => (
  <a {...props} style={{ color: backgroundColor }} href={href}>
    {children}
  </a>
);
