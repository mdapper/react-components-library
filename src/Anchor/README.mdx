---
name: Anchor
route: /Anchor
menu: Components
---

import { Playground, Props } from 'docz';
import { Anchor } from './';

# Anchor

Basic Anchor component

## Usage

<Playground>
  <Anchor>Click here</Anchor>
  <br />
  <Anchor backgroundColor="red">I'm red</Anchor>
</Playground>

## Props

<Props of={Anchor} />
