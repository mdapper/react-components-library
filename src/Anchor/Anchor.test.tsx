import * as React from 'react';
import { render } from '@testing-library/react';

import { Anchor } from './Anchor';

describe('<Anchor />', () => {
  it('renders', () => {
    const { container } = render(<Anchor>Click here</Anchor>);

    expect(container.firstChild).toMatchSnapshot();
  });

  it('renders with backgroundColor red', () => {
    const { container } = render(
      <Anchor backgroundColor="red">I'm red</Anchor>
    );

    expect(container.firstChild).toMatchSnapshot();
  });
});
