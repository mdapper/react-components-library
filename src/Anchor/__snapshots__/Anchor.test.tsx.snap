// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Anchor /> renders 1`] = `
<a
  href="#"
  style="color: blue;"
>
  Click here
</a>
`;

exports[`<Anchor /> renders with backgroundColor red 1`] = `
<a
  href="#"
  style="color: red;"
>
  I'm red
</a>
`;
